### PR TITLE
ci(samples): add nightly resource cleanup

### LINKED ISSUE #789: resource not cleaning up after running samples ITs
We are observing 6000+ test tables and 100+ test datasets that are leftover from running ITs in `java-docs-samples` testing project over time. Some examples of stale tables are listed below:
```
undeleteTable
loadOrcTableFromGcs
addEmptyColumn
addColumnLoadAppend
```

Some examples of stale datasets are shown below:
Screen Shot 2020-10-20 at 11 36 58 AM
We need to think of ways to clean up resources.

### PR DESCRIPTION
**Pull request by stephaniewang526**
Fixes #789

### CODE REVIEW

File: .kokoro/nightly/samples.cfg [hunk 1]
```diff
@@ -36,3 +36,9 @@ env_vars: {
   key: "ENABLE_BUILD_COP"
   value: "true"
 }
+
+# Tell the trampoline which tests to run.
+env_vars: {
+    key: "TRAMPOLINE_BUILD_FILE"
+    value: "github/java-bigquery/.kokoro/run_samples_resource_cleanup.sh"
```
Comment (chingor13) [inline .kokoro/nightly/samples.cfg:43]: Do you want this to still run the samples tests? This will only run the cleanup script.

**Review by chingor13 — CHANGES_REQUESTED** [review]
Also, this file is templatized so make sure it's excluded from synthtool templates.

File: .kokoro/nightly/samples.cfg (hunk 1 above)
Comment (stephaniewang526) [reply .kokoro/nightly/samples.cfg:43]: those are run in the [java8-samples](https://github.com/googleapis/java-bigquery/blob/b1bbddcb87ef5742949e4ec069a4a49b12f9536a/.kokoro/nightly/java8-samples.cfg#L29) and [java11-samples](https://github.com/googleapis/java-bigquery/blob/master/.kokoro/nightly/java11-samples.cfg#L29) configs.

Comment (stephaniewang526) [reply .kokoro/nightly/samples.cfg:43]: Thank you for the reminder -- excluded from synth.py.

**Review by chingor13 — APPROVED** [review]

**Pull request merged**